h/static/scripts/streamer.js

```diff
@@ -155,6 +172,53 @@ function Streamer($rootScope, annotationMapper, groups, session, settings) {
     _connect();
   };
 
+  function applyPendingUpdates() {
+    var updates = Object.values(pendingUpdates).filter(function (ann) {
+      // Ignore updates to annotations that are not in the focused group
+      return ann.group === groups.focused().id;
+    });
+    var deletions = Object.keys(pendingDeletions).map(function (id) {
+      return {id: id};
+    });
+
+    annotationMapper.loadAnnotations(updates);
+    annotationMapper.unloadAnnotations(deletions);
+
+    pendingUpdates = {};
+    pendingDeletions = {};
+  }
+
+  function countPendingUpdates() {
+    return Object.keys(pendingUpdates).length;
```

Comment:
Hmm. This doesn't include pending deletes, which means that deleted annotations will have their reply/share controls disabled with no explanation for the user as to why that is, and no means of removing them from view short of refreshing or waiting for a non-delete update to arrive over the websocket.

Perhaps you've already discussed this, but I reckon pending deletes should be included in this number.